synapse/handlers/federation_event.py

```diff
@@ -1036,6 +1036,9 @@ async def _get_state_ids_after_missing_prev_event(
         # XXX: this doesn't sound right? it means that we'll end up with incomplete
         #   state.
         failed_to_fetch = desired_events - event_metadata.keys()
+        # `event_id` could be missing from `event_metadata` because it's not necessarily a
+        # state event. We've already checked that we've fetched it above.
```

Comment:
```suggestion
        # `event_id` could be missing from `event_metadata` because it's not necessarily
        # a state event. We've already checked that we've fetched it above.
```